More WGLC

jimsch opened the pull request:
Deal with Laurence's github issue #17
Deal with review from Francesa on capabilities.

jimsch committed:
Rename for adoption

Rename for adoption.

jimsch committed:
merge commits

jimsch committed:
Merge remote-tracking branch 'upstream/master'

jimsch committed:
Merge remote-tracking branch 'upstream/master'

jimsch committed:
Merge remote-tracking branch 'upstream/master'

jimsch committed:
Merge remote-tracking branch 'upstream/master'

jimsch committed:
Merge remote-tracking branch 'upstream/master'

jimsch committed:
Move from v2 to v3 vocabulary

Run through the v2v3 converter and then do some edits.

jimsch committed:
Revert "Move from v2 to v3 vocabulary"

This reverts commit 25e6dd85ba786f05b0103d05a96b738f60e71628.

jimsch committed:
Merge remote-tracking branch 'upstream/master'

jimsch committed:
Merge remote-tracking branch 'upstream/master'

jimsch committed:
Merge remote-tracking branch 'upstream/master'

jimsch committed:
More WGLC type comments

Deal with Laurence's github issue #17
Deal with review from Francesa on capabilities.

fpalombini commented on draft-ietf-cose-rfc8152bis-algs.xml:
```diff
@@ -1625,6 +1625,11 @@ COSE_KDF_Context = [
+      </t>
+
       <t>
         Two different types of capabilities are defined:  Capabilities for algorithms and capabilities for key structures. Once defined by registration with IANA, the list capabilities is immutable.
```
"Once defined by registration with IANA, the list capabilities is immutable."

What if a future spec defines that for the same algorithm or key structure another capability is added?

jimsch replied:
You are out of luck in terms of defining the new capability.  It would require a new algorithm or key type.  If it ever changes then any code which uses it for key derivation would be wrong.  I.e. group OSCORE.

fpalombini replied:
I'm ok with that. Maybe add a sentence about it "if you need to define a new capability, you need to register a new algorithm or key type".

fpalombini commented on draft-ietf-cose-rfc8152bis-algs.xml:
```diff
@@ -1625,6 +1625,11 @@ COSE_KDF_Context = [
+
       <t>
         Two different types of capabilities are defined:  Capabilities for algorithms and capabilities for key structures. Once defined by registration with IANA, the list capabilities is immutable.
         As a general rule, the capabilities are going to correspond to algorithm or key fields, but they do not need to do so.
```
suggestion to replace "correspond to" to "be the same for". This might be me but I'm not sure I understand what "corresponds" exactly mean here.

jimsch replied:
Be the same for is not really what I want.  In this context corresponds means maps to.  I think that "maps to" is going to be better.

fpalombini replied:
I much prefer "maps to", thanks.

fpalombini commented on draft-ietf-cose-rfc8152bis-algs.xml:
```diff
@@ -1646,27 +1651,27 @@ COSE_KDF_Context = [
           <li>
```
Can't seem to comment on line 1639 so I do it here:

"The capability structure is an array of values, the order being dependent on the specific algorithm or key."

please clarify "CBOR array of values"

fpalombini commented on draft-ietf-cose-rfc8152bis-algs.xml:
```diff
@@ -1677,23 +1682,33 @@ COSE_KDF_Context = [
         <name>Assignments for Existing Algorithms</name>
         <t>
           For the current set of algorithms in the registry, those in this document as well as those in <xref target="RFC8230"/> and <xref target="I-D.ietf-cose-hash-sig"/>, the capabilities is set to the single entry of the key type that will be accepted.
```
"the capabilities is set to the single entry of the key type that will be accepted."

I know this is quite obvious from the previous section, but I'd really like a sentence stating what is the key type for the algorithm, because that is not part of the COSE Algorithms registry right now. A reference to the previous section would be enough, for example. 

"The key type values covering all algorithms at the time of writing this specification can be found in section 8.1"

or something of the sort.

fpalombini commented on draft-ietf-cose-rfc8152bis-algs.xml:
```diff
@@ -1677,23 +1682,33 @@ COSE_KDF_Context = [
         <t>
           For the current set of algorithms in the registry, those in this document as well as those in <xref target="RFC8230"/> and <xref target="I-D.ietf-cose-hash-sig"/>, the capabilities is set to the single entry of the key type that will be accepted.
-          It is expected other algorithms will have no items or multiple items.
+          It is expected other algorithms could have zero, one, multiple items.
```
s/other/future registered

fpalombini commented on draft-ietf-cose-rfc8152bis-algs.xml:
```diff
@@ -1677,23 +1682,33 @@ COSE_KDF_Context = [
+        <name>Examples</name>
+        <t>
+          In this section a trio of examples are provided.
+          In all three cases the pair of capabilities is always provided as the algoirthm and then the key capabilities.
```
Please specify that the example uses CBOR sequence to format the "pair of capabilities", or format it in a different way (array).

s/algoirthm/algorithm

jimsch replied:
I hate adding the reference just for this.

fpalombini commented on draft-ietf-cose-rfc8152bis-algs.xml:
```diff
@@ -1677,23 +1682,33 @@ COSE_KDF_Context = [
+        <artwork>![CDATA[
+ECDSA with SHA-512 and a P-256 curve:
+
+0x8102820201             / [2][2, 1] /
```
I think the correct way to express this is:
[2][2, 1] ---> [2],[2, 1]

fpalombini commented on draft-ietf-cose-rfc8152bis-algs.xml:
```diff
@@ -1677,23 +1682,33 @@ COSE_KDF_Context = [
+
+ECDH-ES + A256KW with a P-256 curve:
+
+0x8102820201            / [2][2, 1] /
```
[2][2, 1] --> [2],[2, 1]

fpalombini commented on draft-ietf-cose-rfc8152bis-algs.xml:
```diff
@@ -1677,23 +1682,33 @@ COSE_KDF_Context = [
+
+ECDH-ES + A256KW with an X25519 curve:
+
+0x8101820104           / [1][1, 4] /
```
[1][1, 4] --> [1],[1, 4]

jimsch committed:
Deal with Comments on the pull request from Franscesa

jimsch committed:
Respond to John's comments

jimsch committed:
deal with github issues

also deal with text/binary string clarifications.
also deal with OKP definition of x

jimsch committed:
remove extra characters
